packages/orbit-components/src/Itinerary/ItinerarySegment/index.tsx

```diff
@@ -44,29 +44,34 @@ const ItinerarySegment = ({
   onClick,
   banner,
 }: Props) => {
-  const content = React.Children.toArray(children);
+  const content = React.Children.toArray(children) as React.ReactElement[];
 
   const [opened, setOpened] = React.useState(false);
 
   const parts = (
     <Stack direction="column" spacing="none">
       {React.Children.map(children, (el, i) => {
+        if (!React.isValidElement(el)) return null;
         return (
           <ItinerarySegmentContext.Provider
             value={{
               index: i,
               opened,
               toggleOpened: () => setOpened(prev => !prev),
               last: i === React.Children.count(children) - 1,
-              // @ts-expect-error expected
-              isNextHidden: content[i + 1] && content[i + 1].props.hidden,
-              // @ts-expect-error expected
-              isPrevHidden: content[i - 1] && content[i - 1].props.hidden,
+              isNextHidden: !!(
+                content[i + 1] &&
+                content[i + 1].props &&
+                content[i + 1].props.hidden
```

Comment:
I'd personally prefer if it was just
```suggestion
              isNextHidden: Boolean(content[i + 1]?.props?.hidden ?? false)
```

and the same for the one below

Reply:
suggestion is good, except nullish operator, not needed. Still will be evaluated to false

Reply:
True 😃